src/virt.c

```diff
@@ -1166,254 +1152,331 @@ static int lv_init_ignorelists() {
+        ERROR(PLUGIN_NAME " plugin: could not get 'InterfaceFormat' parameter");
+        return -1;
+      }
+
+      if (strcasecmp(format, "name") == 0)
+        interface_format = if_name;
+      else if (strcasecmp(format, "address") == 0)
+        interface_format = if_address;
+      else if (strcasecmp(format, "number") == 0)
+        interface_format = if_number;
+      else {
+        ERROR(PLUGIN_NAME " plugin: unknown InterfaceFormat: %s", format);
+        sfree(format);
+        return -1;
+      }
+
+      sfree(format);
+      continue;
+    } else if (strcasecmp(c->key, "Instances") == 0) {
+      if (cf_util_get_int(c, &nr_instances) != 0) {
+        ERROR(PLUGIN_NAME " plugin: could not get 'Instances' parameter");
+        return -1;
+      }
+
+      if (nr_instances <= 0) {
+        ERROR(PLUGIN_NAME " plugin: Instances <= 0 makes no sense.");
+        return -1;
+      }
+      if (nr_instances > NR_INSTANCES_MAX) {
+        ERROR(PLUGIN_NAME " plugin: Instances=%i > NR_INSTANCES_MAX=%i"
+                          " use a lower setting or recompile the plugin.",
+              nr_instances, NR_INSTANCES_MAX);
+        return -1;
+      }
+
+      DEBUG(PLUGIN_NAME " plugin: configured %i instances", nr_instances);
+      continue;
+    } else if (strcasecmp(c->key, "ExtraStats") == 0) {
+      char *ex_str = NULL;
+
+      if (cf_util_get_string(c, &ex_str) != 0 || ex_str == NULL) {
+        ERROR(PLUGIN_NAME " plugin: could not get 'ExtraStats' parameter");
+        return -1;
+      }
 
-  if (strcasecmp(key, "ExtraStats") == 0) {
-    char *localvalue = strdup(value);
-    if (localvalue != NULL) {
       char *exstats[EX_STATS_MAX_FIELDS];
-      int numexstats =
-          strsplit(localvalue, exstats, STATIC_ARRAY_SIZE(exstats));
-      extra_stats = parse_ex_stats_flags(exstats, numexstats);
-      sfree(localvalue);
+      int numexstats = strsplit(ex_str, exstats, STATIC_ARRAY_SIZE(exstats));
+      int status = parse_ex_stats_flags(&extra_stats, exstats, numexstats);
+      sfree(ex_str);
+      if (status != 0) {
+        ERROR(PLUGIN_NAME " plugin: parsing 'ExtraStats' option failed");
+        return status;
+      }
 
 #ifdef HAVE_JOB_STATS
       if ((extra_stats & ex_stats_job_stats_completed) &&
           (extra_stats & ex_stats_job_stats_background)) {
         ERROR(PLUGIN_NAME " plugin: Invalid job stats configuration. Only one "
                           "type of job statistics can be collected at the same "
                           "time");
-        return 1;
+        return -1;
       }
 #endif
-    }
-  }
 
-  if (strcasecmp(key, "PersistentNotification") == 0) {
-    persistent_notification = IS_TRUE(value);
-    return 0;
-  }
+      /* ExtraStats parsed successfully*/
+      continue;
+    } else if (strcasecmp(c->key, "PersistentNotification") == 0) {
+      if (cf_util_get_boolean(c, &persistent_notification) != 0) {
+        ERROR(PLUGIN_NAME
```

Comment:
Hi @rpv-tomsk,
Yes, I know that cf_util_get.. family of functions has internal reporting - I just wanted to make clearly visible to user that failed option is related to virt plugin, Not sure if unnecessary - prefer more info than less info in case of errors but can remove if required.

Reply:
@rpv-tomsk May we use `DEBUG` message instead of `ERROR` message in these cases ?

Reply:
Why do you need this DEBUG? `cf_util_get_boolean` has ERROR reporting with plugin name included in log messages.

Reply:
We have no particular reason, this was just to output more contextual info to the user. But if you think this is redundant or too much information, we can remove them. Also, in order to be consistent, this means removing such ERROR messages in several other places throughout the plugin (that existed prior to this PR).